plugins/ImageGraph/API.php

```diff
@@ -129,404 +130,453 @@ public function get(
+                $graph->setBackgroundColor($backgroundColor);
+                $graph->setTextColor($textColor);
+                $graph->setException($e);
+                $graph->renderGraph();
             }
 
-            // render graph
-            $graph->renderGraph();
-        } catch (\Exception $e) {
-
-            $graph = new \Piwik\Plugins\ImageGraph\StaticGraph\Exception();
-            $graph->setWidth($width);
-            $graph->setHeight($height);
-            $graph->setFont($font);
-            $graph->setFontSize($fontSize);
-            $graph->setBackgroundColor($backgroundColor);
-            $graph->setTextColor($textColor);
-            $graph->setException($e);
-            $graph->renderGraph();
-        }
+            // restoring get parameters
+            $_GET = $savedGET;
 
-        // restoring get parameters
-        $_GET = $savedGET;
+            switch ($outputType) {
+                case self::GRAPH_OUTPUT_FILE:
+                    if ($idGoal != '') {
+                        $idGoal = '_' . $idGoal;
+                    }
+                    $fileName = self::$DEFAULT_PARAMETERS[$graphType][self::FILENAME_KEY] . '_' . $apiModule . '_' . $apiAction . $idGoal . ' ' . str_replace(',', '-', $date) . ' ' . $idSite . '.png';
+                    $fileName = str_replace(array(' ', '/'), '_', $fileName);
 
-        switch ($outputType) {
-            case self::GRAPH_OUTPUT_FILE:
-                if ($idGoal != '') {
-                    $idGoal = '_' . $idGoal;
-                }
-                $fileName = self::$DEFAULT_PARAMETERS[$graphType][self::FILENAME_KEY] . '_' . $apiModule . '_' . $apiAction . $idGoal . ' ' . str_replace(',', '-', $date) . ' ' . $idSite . '.png';
-                $fileName = str_replace(array(' ', '/'), '_', $fileName);
+                    if (!Filesystem::isValidFilename($fileName)) {
+                        throw new Exception('Error: Image graph filename ' . $fileName . ' is not valid.');
+                    }
 
-                if (!Filesystem::isValidFilename($fileName)) {
-                    throw new Exception('Error: Image graph filename ' . $fileName . ' is not valid.');
-                }
+                    return $graph->sendToDisk($fileName);
 
-                return $graph->sendToDisk($fileName);
+                case self::GRAPH_OUTPUT_PHP:
+                    return $graph->getRenderedImage();
 
-            case self::GRAPH_OUTPUT_PHP:
-                return $graph->getRenderedImage();
+                case self::GRAPH_OUTPUT_INLINE:
+                default:
+                    $pngImgArr[$w]['source'] = $graph->getPngImageData();
+                    $pngImgArr[$w]['id'] = $idSite;
+            }
+        }
+        // render one or more image graphs as HTML page
+        $imageGraphCount = count($pngImgArr);
+        if ($imageGraphCount >= 1) {
+            $this->renderImageGraph($pngImgArr, $imageGraphCount);
+        }
```

Comment:
Sure, doesn't this return an HTML page then when there is only one site (and it was an image before)? It is important that it still returns an image only for one site as it is for example displayed in Android and iOS app. An HTML page would not work there. I might be wrong :)